### PR TITLE
Simplify Button styles and remove isTertiary.

### LINKED ISSUE #23890: Accessibility regression: "Save draft", "Switch to draft", and "Preview" buttons can't be visually identified as user interface controls
**Describe the bug**
In the WordPress 5.5 block editor, the "Save draft", "Switch to draft", and "Preview" buttons have been restyled and in their normal state they now look like plain text.

For accessibility, and I'd say also for a good UI, they should be styled in a way that they can be identified as user interface controls at first glance.

In WordPress 5.4, they weren't perfect either but at least:
- the "Save draft" button, although it missed a shape or an underline, had blue text: blue is a well established convention in WordPress for interactive controls
- same for the "Switch to draft" button
- the "Preview" button looked like a button, which was OK

Screenshots:

save draft preview buttons 5 4

Screenshot 2020-07-12 at 15 02 58

Instead, now in WordPress 5.5 they look like plain text:
- in their normal state, there's no shape or underline to identify them as buttons
- the buttons text color is a dark grey `#1e1e1e` which makes them look as plain text

Screenshots:

save draft preview buttons 5 5

Screenshot 2020-07-12 at 15 02 33

Recommended:
- either restore the blue color and add an underline
- or, preferably, make them look like buttons, because that's what they are

### PR DESCRIPTION
**Pull request by ZebulanStanphill**
## Description
Fixes #23890.

This PR attempts to improve the accessibility of the `Button` component's styles. In general, I tried to clean up some of the messy CSS and make everything more consistent.

The accent color is now used more consistently on buttons: it is now only used for hover, focus, and primary styles. (With the exception of `isLink` buttons, which I haven't touched in this PR.) I always thought it was confusing how the accent color was used as the default outline color of some buttons, but the hover outline color of others.

I've deprecated `isTertiary` and turned it into an alias of `isSecondary` (similar to what happened to `isDefault`). I couldn't find any rhyme or reason as to when one would be chosen over the other, so I figured it would make sense to just merge them.

## How has this been tested?
I've checked just about every button I could find in the UI, both with "Show icon labels" on and with it off, to make sure there are no broken/missing styles.

### Before
image
image
image
image
image
image
image
image
image
image

### After
image
image
image
image
image
image
image
image
image
image

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

**Review by afercia — APPROVED** [review]
Thanks @ZebulanStanphill !
From an accessibility perspective I'd strongly recommend this change, as it fixes what the accessibility team identified as an accessibility regression in WordPress 5.5 (see the issue https://github.com/WordPress/gutenberg/issues/23890).

In the interest of collaboration between teams and for the greater good of the project, I do realize this needs design feedback 🙂  Approving from an a11y perspective.

**Comment by joedolson** [comment]
This looks great from my perspective, too; although I recognize that I'm basically just seconding @afercia's accessibility approval. ;)

**Review by youknowriad — CHANGES_REQUESTED** [review]
Let's get this a proper design review, I'm pretty sure this is a regression in terms of the buttons importance hierarchy

**Comment by enriquesanchez** [comment]
Thank you for working in this @ZebulanStanphill!

I understand the need to address #23890, but putting my designer hat on I have some different feedback.

At first glance, these buttons look like input fields to me. You can see how there's little to no visual difference between the proposed style for buttons and the 'Add tags' field:

Screen Shot 2020-09-28 at 11 04 20

This is also affecting the hierarchy of buttons and options. While this may not be an issue when there are just a couple of buttons, I find it very hard to parse something like this:

94205841-ba6e8000-fe89-11ea-874e-850bf1b43321

We need a balance that makes it clear these are buttons but that also not overwhelm a whole other set of users.

We're also not accounting for visual inconsistencies across the UI, like icon buttons:

Screen Shot 2020-09-28 at 11 10 44

or disabled states like the 'Redo' button over here:

94206117-4e404c00-fe8a-11ea-99ae-ffa6e16324f1

--- 

Overall I think your exploration is incredibly useful, as it allows us to find these nuanced cases that we need to be aware of when exploring a solution.

**Comment by mariohamann** [comment]
The three button styles seem to follow Material Design: https://material.io/components/buttons#hierarchy-and-placement

I understand the PR from a11y perspective to identify every button directly as a button. However,  I don't think merging the second and third level is the way to go...

**Comment by ZebulanStanphill** [comment]
The points about icon button and disabled button styles are both very valid. The quick answer would be to just add borders to all of them, but in the case of icon buttons, that seems a bit too much in the top bar.

Thinking about it some more, I think there needs to be two different classes of button styles: standard and toolbar. Standard buttons would always have an outline, while toolbar buttons wouldn't, since their usage in a toolbar should be enough to make it clear that they're buttons. How does this sound?

There's a bit of a problem with the top bar, however: it's actually not a toolbar. Or at least not all of it. The left half is a toolbar, but the right half isn't.

As for buttons looking like input fields, that's another good point. But both buttons and input fields ought to have clearly defined borders. So we need to find a different way to distinguish the two via different shaping. Any ideas?

@mariohamann It's worth remembering that Material Design patterns were not designed with accessibility in mind.

The problem with using buttons that consist only of accent-colored text is that accessible designs must account for colorblind users. There needs to be distinct contrast and shaping for all variations. Again, there must be a recognizable difference in shape/contrast. Additionally, blue-text buttons look a lot like links, which they aren't.

**Comment by afercia** [comment]
> It's worth remembering that Material Design patterns were not designed with accessibility in mind.

+1 to this. Keeping using Material Design as a reference in a project that aims to be accessible isn't that productive 🙂

**Comment by kellychoffman** [comment]
Agree its not productive to compare to Material in this case, so let's just stick with the issue we are trying to solve. The buttons currently look like plain text but this particular solution will have them looking like input fields (and introduces additional a11y issues as brought up by @enriquesanchez.) Issues that I think are blockers. I think this was a solid attempt but needs to go back to the drawing board to make the UI both accessible and intuitive. (Underlines immediately comes to mind but am open to other options.)

**Comment by ZebulanStanphill** [comment]
Agreed, this needs some more discussion. Let's continue in #23890. If anyone has any ideas, please share them there.

**Comment by afercia** [comment]
> We need a balance that makes it clear these are buttons but that also not overwhelm a whole other set of users.

I do understand this concern. @enriquesanchez as designer and also accessibility team member and representative, do you have any concrete proposal to find a better balance?

**Comment by ZebulanStanphill** [comment]
Since it's considerably out-of-date, and there's no final design solution in sight, I'm closing this PR.

I still think there's a need for a clearer and more consistent visual distinction between links, buttons, and `<input>`s. However, precisely what each should look like remains an open question. Until that question is answered, no progress can be made here.

I should also note that I may have gone overboard with putting outlines on **every** text button. I think the ones in toolbars can probably go without them, since they're known to be clickable by context, in the same way the icon buttons are known to be clickable.

Please continue discussion in #23890.
